src/doc/dev_manual/Spack.rst

```diff
@@ -0,0 +1,362 @@
+
+    spack spec visit ^python+shared ^mesa+opengl
+
+To build VisIt_ with the default compiler. ::
+
+    spack install visit ^python+shared ^mesa+opengl
+
+To build VisIt_ with a specific compiler. ::
+
+    pack install visit%gcc@11.2.0 ^python+shared ^mesa+opengl
+
+To build VisIt_ and specify a specific version of a dependent library. ::
+
+    pack install visit%gcc@11.2.0 ^python+shared ^mesa+opengl ^llvm@11.0.1
+
+To uninstall a package (the ``--dependents`` uninstalls all the packages that depend on the uninstalled packages). ::
+
+    spack uninstall --dependents visit vtk
+
+To uninstall all the packages. ::
+
+    spack uninstall --all
+
+The package files for a package are located in ``var/spack/repos/builtin/package``.
+Here are some examples. ::
+
+    ./var/spack/repos/builtin/packages/visit
+    ./var/spack/repos/builtin/packages/vtk
+
+The installed packages are stored in ``opt/spack/<architecture>/<compiler>``.
+Here are some examples. :: 
+
+    opt/spack/cray-sles15-zen2/gcc-11.2.0/hdf5-1.8.22-c3djozhlmrvy7wpu46f36qeakemiactw
+    opt/spack/cray-sles15-zen2/gcc-11.2.0/cmake-3.14.7-nnahgnkkl2d2ty2us46we75pnjepci35
+
+The spack environment files
+---------------------------
+
+Spack uses two files to control the environment on a system. They are the ``compilers.yaml`` file and the ``packages.yaml`` file.
+
+The ``compilers.yaml`` file is used to specify information about compilers on a system.
+The ``spack compilers`` command, as mentioned earlier, will create one for you with all the compilers on the system.
+You can then customize it.
+The complete documentation on the ``compilers.yaml`` file can be found `here <https://spack.readthedocs.io/en/latest/getting_started.html#compiler-configuration>`__.
+
+The ``packages.yaml`` file is used to specify information about external packages on a system.
+By default, spack will want to build everything from scratch for your system.
+If some of the packages are already installed on the system you can use those by listing them in a ``packages.yaml`` file.
+Typically, you will want to use an external MPI library on most HPC systems.
+The ``spack externals find`` command will create an initial ``packages.yaml`` file for you.
+The ``spack externals find`` command is non-destructive and will append to an existing ``packages.yaml`` file.
+You can then customize it.
+The complete documentation on the ``packages.yaml`` file can be found `here <https://spack.readthedocs.io/en/latest/build_settings.html>`__.
+
+These files are stored in your ``~/.spack`` directory. ::
+
+    .spack
+    .spack/<platform>
+    .spack/<platform>/compilers.yaml
+    .spack/packages.yaml
+
+The VisIt_ repository at GitHub contains ``compilers.yaml`` and ``packages.yaml`` files for popular systems.
+
+Here are the ``compilers.yaml`` and ``packages.yaml`` files for ``spock.olcf.ornl.gov`` for VisIt_.
+
+``compilers.yaml`` ::
```

Comment:
Can we just use the sphinx include declaration here instead of cutting/pasting actual code?

Reply:
Apologies...I guess I had assumed these files were already under revision control here.